addons/mrp/static/src/js/mrp_field_one2many_with_copy.js

```diff
@@ -0,0 +1,82 @@
+odoo.define('mrp.MrpFieldOne2ManyWithCopy', function (require) {
+
+"use strict";
+    
+var FieldOne2Many = require('web.relational_fields').FieldOne2Many;
+var ListRenderer = require('web.ListRenderer');
+var fieldRegistry = require('web.field_registry');
+var core = require('web.core');
+var _t = core._t;
+    
+var MrpFieldOne2ManyWithCopyListRenderer = ListRenderer.extend({
+
+    /**
+     * @override
+     */
+    init: function (parent, state, params) {
+        this._super.apply(this, arguments);
+        this.creates.push({
+            string: parent.nodeOptions.copy_text,
+            context: '',
+        });
+    },
+});
+
+var MrpFieldOne2ManyWithCopy = FieldOne2Many.extend({
+    
+    /**
+     * @override
+     */
+    init: function (parent, name, record, options) {
+        this._super.apply(this, arguments);
+        this.nodeOptions = _.defaults(this.nodeOptions, {
+            copy_text: _t('Copy line'),
+        });
+    },
+    /**
+     * @override
+     */
+    _getRenderer: function () {
+        if (this.view.arch.tag === 'tree') {
+            return MrpFieldOne2ManyWithCopyListRenderer;
+        }
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * @override
+     */
+    _openFormDialog: function (params) {
+        if('id' in params || params.context === undefined) {
+            return this._super.apply(this, arguments);
+        }
+        let parentIsNew = this.getParent().state.res_id === undefined;
+        let parentHasChanged = this.getParent().state.isDirty();
+        if(parentIsNew || parentHasChanged) {
```

Comment:
```suggestion
        if ('id' in params || params.context === undefined) {
            return this._super.apply(this, arguments);
        }
        let parentIsNew = this.getParent().state.res_id === undefined;
        let parentHasChanged = this.getParent().state.isDirty();
        if (parentIsNew || parentHasChanged) {
```
(sorry, I'm a little bit picky with this kind of thing :grimacing: )